src/cita/itemWrapper.ts

```diff
@@ -63,150 +69,254 @@ export default class ItemWrapper {
 	// this.place; // for books?
 
 	// UUIDs
-	get doi(): string | undefined {
-		return this.getPID("DOI");
+	get doi(): DOI | undefined {
+		return this.getPID("DOI")?.id as DOI;
 	}
 
-	set doi(doi: string) {
+	set doi(doi: DOI) {
 		this.setPID("DOI", doi);
 	}
 
 	get isbn(): string | undefined {
-		return this.getPID("ISBN");
+		return this.getPID("ISBN")?.id;
 	}
 
 	set isbn(isbn: string) {
 		this.setPID("ISBN", isbn);
 	}
 
 	get qid(): QID | undefined {
-		return this.getPID("QID") as QID;
+		return this.getPID("QID")?.id as QID;
 	}
 
 	set qid(qid: QID) {
 		this.setPID("QID", qid);
 	}
 
-	// OpenCitations Corpus Internal Identifier
-	get occ(): string | undefined {
-		return this.getPID("OCC");
+	get omid(): OMID | undefined {
+		return this.getPID("OMID")?.id as OMID;
 	}
 
-	set occ(occ: string) {
-		this.setPID("OCC", occ);
+	set omid(omid: OMID) {
+		this.setPID("OMID", omid);
 	}
 
-	get url() {
+	get url(): string | null {
 		const url = this.item.getField("url");
 		return (
 			url ||
 			this.getPidUrl("QID") ||
 			this.getPidUrl("DOI") ||
-			this.getPidUrl("OCC")
+			this.getPidUrl("OMID")
 		);
 	}
 
+	static readonly allTypesToShow: PIDType[] = [
+		"DOI",
+		"ISBN",
+		"QID",
+		"OMID",
+		"arXiv",
+		"OpenAlex",
+		"CorpusID",
+		// Don't show PMID or PMCID because we can't fetch citations from them
+	];
+
 	getPIDTypes() {
-		const allTypes: PIDType[] = ["DOI", "ISBN", "QID", "OCC"];
 		const pidTypes: PIDType[] = [];
-		for (const type of allTypes) {
-			// don't need this because we enforce that it's uppercase already
-			// type = type.toUpperCase();
+		for (const type of ItemWrapper.allTypesToShow) {
 			switch (type) {
 				case "DOI":
```

Comment:
I know that Zotero is adamant that DOIs don't exist for books, yet they often do and are more reliable than ISBNs in my experience. I've updated `getPID`, `setPID`, and `fetchPID` to better support DOIs for "unsupported" items. Now, should this be shown in the PID rows by pushing `"DOI"` here regardless of its supported status?
```suggestion
				case "DOI":
                    pidTypes.push(type);
```

Reply:
Maybe it would make sense to still be a valid field that you can add, but not to show by default for books (unless the book has a DOI when you add it)? Does that make sense?

Reply:
In other words: DOI freely gettable, settable and fetchable (as is now the case), but we only display it if it's a valid field for the type or if it has a value?

Reply:
Yeah that sounds good!